[MM-17703] Diff config files

threepwood-mm opened the pull request:
#### Summary

This change adds a make task that allows getting changes in default config files between releases.

To run it, use the following command:
```
make diff-config FROM=5.11.0 TOA=5.13.2
```

This will display the difference between release `5.11.0` and `5.13.2`. 

![1566192689](https://user-images.githubusercontent.com/54032291/63241270-668b7380-c253-11e9-93dc-e002dfeb2622.png)

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/11859

threepwood-mm commented:
I'd like to know if the current output (`diff -y`) is enough or there's a different preference.

mkraft commented:
> I'd like to know if the current output (`diff -y`) is enough or there's a different preference.

@threepwood-mm I'm wondering if it would be useful for the non-devs to include a sample output?

threepwood-mm commented:
Added a sample of the `diff -y`.

jasonblais commented:
Thanks @threepwood-mm! I'm curious how changes to default values would be displayed?

threepwood-mm commented:
> Thanks @threepwood-mm! I'm curious how changes to default values would be displayed?

It shows a `|`, like this:
![1566216990](https://user-images.githubusercontent.com/54032291/63264733-167cd300-c28c-11e9-83a0-c30c77a54a57.png)

mkraft commented on scripts/diff-config.sh:
```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env bash
+if [ -z "$FROM" ]
+then
+  echo "Missing FROM version. Usage: make diff-config FROM=v1.2.3 TO=v1.2.3"
```
```suggestion
  echo "Missing FROM version. Usage: make diff-config FROM=5.13.2 TO=5.14.0"
```

mkraft replied:
```suggestion
  echo "Missing FROM version. Usage: make diff-config FROM=5.13.3 TO=5.14.0"
```

mkraft commented on scripts/diff-config.sh:
```diff
@@ -0,0 +1,44 @@
+
+if [ -z "$TO" ]
+then
+  echo "Missing TO version. Usage: make diff-config FROM=v1.2.3 TO=v1.2.3"
```
```suggestion
  echo "Missing TO version. Usage: make diff-config FROM=5.13.2 TO=5.14.0"
```

threepwood-mm commented:
I've added your suggestion, but kept `1.2.3` to make it clearer it's a placeholder. Do you prefer to show a valid usage example instead, like the one you proposed?
Also added the `jq` check, I think it makes a lot of sense.

mkraft commented:
> I've added your suggestion, but kept `1.2.3` to make it clearer it's a placeholder. Do you prefer to show a valid usage example instead, like the one you proposed?

Yeah, 1.2.3 is cool with me. Only thought is shouldn't the `FROM` and `TO` example values be different?

threepwood-mm commented:
> > I've added your suggestion, but kept `1.2.3` to make it clearer it's a placeholder. Do you prefer to show a valid usage example instead, like the one you proposed?
> 
> Yeah, 1.2.3 is cool with me. Only thought is shouldn't the `FROM` and `TO` example values be different?

I've no preference on that. For me, `1.2.3` is a placeholder that tells the expected format of the field and the fact that `FROM` and `TO` matches doesn't matter. Is like, for example, if they were integer fields, just showing `0` for both would suffice to make clear what's expecting.
But again, I've no preference, two different placeholder values also make sense to me.

mkraft commented:
> > > I've added your suggestion, but kept `1.2.3` to make it clearer it's a placeholder. Do you prefer to show a valid usage example instead, like the one you proposed?
> > 
> > 
> > Yeah, 1.2.3 is cool with me. Only thought is shouldn't the `FROM` and `TO` example values be different?
> 
> I've no preference on that. For me, `1.2.3` is a placeholder that tells the expected format of the field and the fact that `FROM` and `TO` matches doesn't matter. Is like, for example, if they were integer fields, just showing `0` for both would suffice to make clear what's expecting.
> But again, I've no preference, two different placeholder values also make sense to me.

My preference would be to have a `TO` value that is ahead of the `FROM` to remove any possible ambiguity.

threepwood-mm commented:
@grundleborg @amyblais do you have a moment to review this one?

hanzei commented:
@lindalumitchell Would you please request a QA review for this PR?

lindalumitchell commented:
Thanks @hanzei, looks good to me to proceed. Removed QA Review label.

hanzei commented:
Thanks you very much for the contribution @threepwood-mm :tada:
